academy/ibc-dev/ibcapp_checkers.md

```diff
@@ -0,0 +1,525 @@
+    } else {
+        panic(fmt.Sprintf(types.ErrWinnerNotParseable.Error(), storedGame.Winner))
+    }
+    return winnerAddress, loserAddress
+}
+
+func (k *Keeper) MustRegisterPlayerWin(ctx sdk.Context, storedGame *types.StoredGame) {
+    winnerAddress, loserAddress := getWinnerAndLoserAddresses(storedGame)
+    k.board.MustAddWonGameResultToPlayer(ctx, winnerAddress)
+    k.board.MustAddLostGameResultToPlayer(ctx, loserAddress)
+}
+
+func (k *Keeper) MustRegisterPlayerForfeit(ctx sdk.Context, storedGame *types.StoredGame) {
+    winnerAddress, loserAddress := getWinnerAndLoserAddresses(storedGame)
+    k.board.MustAddWonGameResultToPlayer(ctx, winnerAddress)
+    k.board.MustAddForfeitedGameResultToPlayer(ctx, loserAddress)
+}
+
+```
+
+You can see that the checkers module will need to access the leaderboard methods, like `k.board.MustAddWonGameResultToPlayer(...)`.
+
+First, you need to write those functions. Create a `x/leaderboard/keeper/player_info_handler.go` file with the following code: 
+
+```golang
+package keeper
+
+import (
+    "github.com/b9lab/checkers/x/leaderboard/types"
+    sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func mustAddDeltaGameResultToPlayer(
+    k *Keeper,
+    ctx sdk.Context,
+    player sdk.AccAddress,
+    wonDelta uint64,
+    lostDelta uint64,
+    forfeitedDelta uint64,
+) (playerInfo types.PlayerInfo) {
+    playerInfo, found := k.GetPlayerInfo(ctx, player.String())
+    if !found {
+        playerInfo = types.PlayerInfo{
+            Index:          player.String(),
+            WonCount:       0,
+            LostCount:      0,
+            ForfeitedCount: 0,
+            DateUpdated:    ctx.BlockTime().UTC().Format(types.TimeLayout),
+        }
+    }
+    playerInfo.WonCount += wonDelta
+    playerInfo.LostCount += lostDelta
+    playerInfo.ForfeitedCount += forfeitedDelta
+    k.SetPlayerInfo(ctx, playerInfo)
+    return playerInfo
+}
+
+func (k *Keeper) MustAddWonGameResultToPlayer(ctx sdk.Context, player sdk.AccAddress) types.PlayerInfo {
+    return mustAddDeltaGameResultToPlayer(k, ctx, player, 1, 0, 0)
+}
+
+func (k *Keeper) MustAddLostGameResultToPlayer(ctx sdk.Context, player sdk.AccAddress) types.PlayerInfo {
+    return mustAddDeltaGameResultToPlayer(k, ctx, player, 0, 1, 0)
+}
+
+func (k *Keeper) MustAddForfeitedGameResultToPlayer(ctx sdk.Context, player sdk.AccAddress) types.PlayerInfo {
+    return mustAddDeltaGameResultToPlayer(k, ctx, player, 0, 0, 1)
+}
+```
+
+For the code above to function, you need to define `TimeLayout` in the `x/leaderboard/types/keys.go`. Add the following piece of code at the end of the file:
+
+```golang
+const (
+    TimeLayout              = "2006-01-02 15:04:05.999999999 +0000 UTC"
+    LeaderboardWinnerLength = uint64(100)
+)
+```
+
+Now it is time to allow the checkers module access to the leaderboard module. Therefore, look for the `app.CheckersKeeper` in `app/app.go` and modify it in order to include `app.LeaderboardKeeper`:
```

Comment:
This is a nice consequence of the approach you went for, allowing to review passing capabilities to another module :)